### PR TITLE
Test get msg after index migration

### PR DESCRIPTION
**Pull request by Weistrand**
Adds a data dir from a server with index version 2 to spec resources, the data dir has a queue with a message. A spec starts a server from this dir and thereafter migrate index version and then try to get the message. This spec was built after a bug in the migration process. See [slack thread](https://84codes.slack.com/archives/CKPHUKCF6/p1668493121825119) for more info.

### CODE REVIEW

File: spec/spec_helper.cr [hunk 1]
```diff
@@ -11,6 +11,9 @@ require "uri"
 
 FileUtils.rm_rf("/tmp/lavinmq-spec")
 
+FileUtils.rm_rf("./spec/resources/data_dir_index_v2_new")
+FileUtils.cp_r("./spec/resources/data_dir_index_v2", "./spec/resources/data_dir_index_v2_new")
```
Comment (carlhoerberg) [inline spec/spec_helper.cr:15]: Have this as a `before_each` in the `context` block in Storage spec ?

**Review by carlhoerberg — CHANGES_REQUESTED** [review]

File: spec/storage_spec.cr [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+      close_servers
+      TestHelpers.create_servers("./spec/resources/data_dir_index_v2_new")
+      with_channel do |ch|
+        q = ch.queue("queue")
+        s.vhosts["/"].queues["queue"].as(LavinMQ::DurableQueue)
+        q.get(no_ack: true).try(&.body_io.to_s).should eq("message")
+      end
+      close_servers
+      FileUtils.rm_rf("./spec/resources/data_dir_index_v2_new")
+      TestHelpers.setup
```
[the remark this answers is not in the document]
Comment (Weistrand) [reply spec/storage_spec.cr:19]: [64f7dd5](https://github.com/cloudamqp/lavinmq/pull/423/commits/64f7dd5c488800592e775ca4489a0832a21acd99)

File: spec/spec_helper.cr (hunk 1 above)
Comment (Weistrand) [reply spec/spec_helper.cr]: [d731eb4](https://github.com/cloudamqp/lavinmq/pull/423/commits/d731eb491e10f6fcf14aecd34caa32e2ba55b1dd) and [5a761e9](https://github.com/cloudamqp/lavinmq/pull/423/commits/5a761e9d28442de860168b7bc4d46bd9b759cc92)

**Comment by carlhoerberg** [comment]
@Weistrand wdyt of https://github.com/cloudamqp/lavinmq/pull/423/commits/95802f2030c46217fad7509327f5d1160cdfbc4f ?

**Comment by Weistrand** [comment]
> @Weistrand wdyt of [95802f2](https://github.com/cloudamqp/lavinmq/commit/95802f2030c46217fad7509327f5d1160cdfbc4f) ?

Looks good!

**Pull request merged**